server/Dockerfile

```diff
@@ -0,0 +1,76 @@
+###################################
+# STAGE 1 - Install Server
+###################################
+FROM python:3.8-slim-buster as server
+# Setup variables
+ENV GUNICORN_DIRECTORY=/server/gunicorn \
+    GUNICORN_WORKERS=3 \
+    GUNICORN_WORKER_CLASS=gevent \
+    GUNICORN_BACKLOG=2048 \
+    GUNICORN_BIND=0.0.0.0:80 \
+    SERVER_MODULE=main:app \
+    SERVER_DIRECTORY=/server \
+    CLIENT_DIRECTORY=/client
+
+# Setup gunicorn config
+WORKDIR ${GUNICORN_DIRECTORY}
+COPY server/gunicorn .
+
+# Navigate to server directory
+WORKDIR ${SERVER_DIRECTORY}
+
+# Install server dependencies
+RUN pip install pipenv
+COPY server/Pipfile* ./
+RUN pipenv install --system --deploy
+
+# Copy remaining server source
+COPY server .
+
+
+###################################
+# STAGE 2 - Development
+###################################
+FROM server as development
+
+# Run server
+CMD gunicorn --config ${GUNICORN_DIRECTORY}/gunicorn.conf.py --log-config ${GUNICORN_DIRECTORY}/logging.conf \
+    --chdir ${SERVER_DIRECTORY} ${SERVER_MODULE}
+
+
+######################################
+# STAGE 3a - Production (Build Client)
+######################################
+FROM node:12.18.3-buster-slim as build-client
+# Setup variables
+ENV CLIENT_DIRECTORY=/client
```

Comment:
I think this is an unnecessary overwrite? It's set up above as well, with SERVER_DIRECTORY, which is not being re-set.

Reply:
No, this is a different image (see the FROM is different). SERVER_DIRECTORY is not needed in this image.

Reply:
got it. might make sense to remove CLIENT_DIRECTORY from above then because it is not used up there it appears?

Reply:
ohhhh oh oh oh wait nvm i get it now